--- origsrc/httrack-3.42.2/src/htsmodules.c	2008-03-16 16:26:22.000000000 -0500
+++ src/httrack-3.42.2/src/htsmodules.c	2008-03-20 17:22:19.031250000 -0500
@@ -267,6 +267,12 @@
       void* handle;
 #ifdef _WIN32
       handle = LoadLibraryA((char*)"ssleay32");
+#elif defined(__CYGWIN__)
+      /* We are compatible with 0.9.7/8 and potentially above */
+      handle = dlopen("cygssl-0.9.8.dll", RTLD_LAZY);
+      if (handle == NULL) {
+        handle = dlopen("cygssl-0.9.7.dll", RTLD_LAZY);
+      }
 #else
       /* We are compatible with 0.9.6/7/8/8b and potentially above */
       handle = dlopen("libssl.so.0.9.8b", RTLD_LAZY);  /* added 8b release too (Debarshi Ray) */
