--- origsrc/httrack-3.42.2/src/htsglobal.h	2008-03-16 16:26:22.000000000 -0500
+++ src/httrack-3.42.2/src/htsglobal.h	2008-03-20 17:22:19.000000000 -0500
@@ -209,7 +209,7 @@
 #define HTS_USESWF 1
 #endif
 
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__CYGWIN__)
 #else
 #define __cdecl
 #endif
@@ -370,7 +370,7 @@
 /* IPV4, IPV6 and various unified structures */
 #define HTS_MAXADDRLEN 64
 
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__CYGWIN__)
 #else
 #define __cdecl 
 #endif
--- origsrc/httrack-3.47.27/src/mmsrip/mms.h	2013-06-28 12:45:58.000000000 -0500
+++ src/httrack-3.47.27/src/mmsrip/mms.h	2013-10-24 21:17:28.528361300 -0500
@@ -40,7 +40,7 @@ typedef signed long int ssize_t;
 #define SHUT_RDWR SD_BOTH
 #endif
 
-#if defined(__CYGWIN__) || defined(_WIN32)
+#if defined(_WIN32)
 typedef unsigned char uint8_t;
 #ifndef __uint32_t_defined
 #define __uint32_t_defined
